google-cloud-notification/src/test/java/com/google/cloud/notification/it/ITSystemTest.java

```diff
@@ -0,0 +1,136 @@
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+public class ITSystemTest {
+
+  private static RemoteStorageHelper remoteStorageHelper;
+  private static TopicAdminClient topicAdminClient;
+  private static Notification notificationService;
+  private static Storage storageService;
+
+  private static final Logger log = Logger.getLogger(ITSystemTest.class.getName());
+  private static final String BUCKET = RemoteStorageHelper.generateBucketName();
+  private static final String NAME_SUFFIX = UUID.randomUUID().toString();
+  private static String projectId;
+
+  @Rule
+  public Timeout globalTimeout = Timeout.seconds(300);
+
+  @BeforeClass
+  public static void beforeClass() throws NoSuchAlgorithmException, InvalidKeySpecException, IOException {
+    remoteStorageHelper = RemoteStorageHelper.create();
+    topicAdminClient = TopicAdminClient.create();
+    storageService = remoteStorageHelper.getOptions().getService();
+    notificationService = new DefaultNotificationFactory().create(remoteStorageHelper.getOptions());
+    storageService.create(BucketInfo.of(BUCKET));
+    projectId = ServiceOptions.getDefaultProjectId();
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    topicAdminClient.close();
+    if (notificationService != null) {
+      boolean wasDeleted = RemoteStorageHelper.forceDelete(storageService, BUCKET, 5, TimeUnit.SECONDS);
+      if (!wasDeleted && log.isLoggable(Level.WARNING)) {
+        log.log(Level.WARNING, "Deletion of bucket {0} timed out, bucket is not empty", BUCKET);
+      }
+    }
+  }
+
+  private String formatForTest(String resourceName) {
+    return resourceName + "-" + NAME_SUFFIX;
+  }
+
+  @Test
+  public void testNotifications() {
+    // Use Pubsub to create a Topic.
+    final TopicName topic = TopicName.create(projectId, formatForTest("testing-topic-foo"));
+    topicAdminClient.createTopic(topic);
+
+    Policy policy = topicAdminClient.getIamPolicy(topic.toString());
+    Binding binding =
+        Binding.newBuilder().setRole("roles/owner").addMembers("allAuthenticatedUsers").build();
+    Policy newPolicy =
+        topicAdminClient.setIamPolicy(
+            topic.toString(), policy.toBuilder().addBindings(binding).build());
+    assertTrue(newPolicy.getBindingsList().contains(binding));
+    String permissionName = "pubsub.topics.get";
+    List<String> permissions =
+        topicAdminClient
+            .testIamPermissions(topic.toString(), Collections.singletonList(permissionName))
+            .getPermissionsList();
+    assertTrue(permissions.contains(permissionName));
+
+    // Use Storage API to create a Notification on that Topic.
+    NotificationInfo notification = notificationService.createNotification(BUCKET,  NotificationInfo.of(topic));
+    assertNotNull(notification);
+    Set<NotificationInfo> notifications = Sets.newHashSet(notificationService.listNotifications(BUCKET));
```

Comment:
You can just keep this as a list; it also has a `contains` method. There's only 1 element anyway right?

Reply:
yep!